packages/react-vapor/src/components/select/SelectConnected.tsx

```diff
@@ -196,16 +196,16 @@ export class SelectConnected extends React.PureComponent<ISelectProps & ISelectS
             const actives = _.chain(this.props.items)
                 .filter(
                     (item: IItemBoxProps) =>
-                        (!item.hidden && !this.props.multi && !item.disabled) ||
-                        !_.contains(this.props.selectedValues, item.value)
+                        !item.hidden &&
+                        (!this.props.multi || !_.contains(this.props.selectedValues, item.value)) &&
+                        !item.disabled
                 )
                 .value();
             const active = actives[mod(this.props.active, actives.length)];
             if (active) {
                 this.props.onSelectValue(active.value, this.props.multi);
-                this.onToggleDropdown(e);
             }
-        } else if (_.contains([keyCode.enter, keyCode.downArrow, keyCode.upArrow], e.keyCode) && !this.props.isOpened) {

```

Comment:
I think you can keep the "new" logic here

Reply:
Done, works well!